src/main/scala/JavaVersionCheckPlugin.scala

```diff
@@ -22,21 +22,21 @@ object JavaVersionCheckPlugin extends sbt.AutoPlugin {
     javaVersionPrefix in javaVersionCheck := defaultJavaVersionPrefix,
     javaVersionCheck := {
       val log = streams.value.log
-      val javac = (compileInputs in (Compile, compile)).value.compilers.javac
+      val javac = (compileInputs in (Compile, compile)).value.compilers.javaTools().javac()
       JavaVersionCheck((javaVersionPrefix in javaVersionCheck ).value, javac, log)
     },
-    // we hook onto deliverConfiguration to run the version check as early as possible,
+    // we hook onto publishConfiguration to run the version check as early as possible,
     // before we actually do anything. But we don't want to require the version check
     // just for compile.
-    deliverConfiguration := {
+    publishConfiguration := {
```

Comment:
I had to use another task since `deliverConfiguration` seems to be gone in sbt 1.x

Reply:
ok